src/Storage/Telemetry/TelemetryEnrichingMiddleware.cs

```diff
@@ -0,0 +1,149 @@
+                }
+            },
+            {
+                AltinnCoreClaimTypes.PartyID,
+                static (claim, activity) =>
+                {
+                    activity.SetTag("user.party.id", claim.Value);
+                }
+            },
+            {
+                AltinnCoreClaimTypes.AuthenticationLevel,
+                static (claim, activity) =>
+                {
+                    activity.SetTag("user.authentication.level", claim.Value);
+                }
+            },
+            {
+                AltinnCoreClaimTypes.Org,
+                static (claim, activity) =>
+                {
+                    activity.SetTag("user.application.owner.id", claim.Value);
+                }
+            },
+            {
+                AltinnCoreClaimTypes.OrgNumber,
+                static (claim, activity) =>
+                {
+                    activity.SetTag("user.organization.number", claim.Value);
+                }
+            },
+            {
+                "authorization_details",
+                static (claim, activity) =>
+                {
+                    SystemUserClaim claimValue = JsonSerializer.Deserialize<SystemUserClaim>(claim.Value);
+                    activity.SetTag("user.system.id", claimValue?.Systemuser_id[0] ?? null);
+                    activity.SetTag("user.system.owner.number", claimValue?.Systemuser_org.ID ?? null);
+                }
+            },
+        };
+
+        return actions.ToFrozenDictionary();
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TelemetryEnrichingMiddleware"/> class.
+    /// </summary>
+    /// <param name="next">The next middleware in the pipeline.</param>
+    /// <param name="logger">The logger instance.</param>
+    /// <param name="generalSettings">Configuration object used to hold general settings for the storage application.</param>
+    public TelemetryEnrichingMiddleware(RequestDelegate next, ILogger<TelemetryEnrichingMiddleware> logger, IOptions<GeneralSettings> generalSettings)
+    {
+        _next = next;
+        _logger = logger;
+        _disableTelemetryForMigration = generalSettings.Value.DisableTelemetryForMigration;
+    }
+
+    /// <summary>
+    /// Invokes the middleware to process the HTTP context.
+    /// </summary>
+    /// <param name="context">The HTTP context.</param>
+    public async Task InvokeAsync(HttpContext context)
+    {
+        var activity = context.Features.Get<IHttpActivityFeature>()?.Activity;
+        if (activity is null || !context.Request.Path.ToString().Contains("storage/api/"))
+        {
+            await _next(context);
+            return;
+        }
+
+        if (_disableTelemetryForMigration && context.Request is not null && context.Request.Path.ToUriComponent().StartsWith("/storage/api/v1/migration", StringComparison.OrdinalIgnoreCase))
+        {
+            await _next(context);
```

Comment:
Is this missing a return?

Reply:
I suspect that the current IdentityTelemetryFilter.cs implementation is wrong when it skips a call to Next.Process().